Handle new message for unterminated lists on MRI 2.7

casperisfine opened the pull request:
Ref: https://github.com/pry/pry/pull/2087

Pry test suite breaks on MRI 2.7 previews because of this.

The error message for `issue = %W/` changed, before:

```
(eval):2: syntax error, unexpected tSTRING_END, expecting tSTRING_CONTENT or tSTRING_DBEG or tSTRING_DVAR or ' '
```

After:

```
((eval):2: unterminated list meets end of file)
```

@kyrylo @banister

kyrylo commented:
Oh wait, I actually do.

kyrylo commented:
Seems like this gem needs some maintenance first. Lots of failing specs that seem to be unrelated to this change.

casperisfine commented:
It's mostly old rubies failing to install on Travis. I can make another PR for that if you want.

casperisfine commented:
@kyrylo done.

https://github.com/banister/method_source/pull/60 fixes master CI and add newer rubies.

I rebased this PR on top of it and added 2.7 as well.

It's now green across the board.

kyrylo commented:
Cool, I merged https://github.com/banister/method_source/pull/60. Could you rebase again?

casperisfine commented:
@kyrylo done.

barrettkingram commented:
@kyrylo @banister Reminder that this still needs to be released, especially because ruby 2.7 is officially out now.
